src/browser/serial.js

```diff
@@ -227,17 +227,29 @@ function SerialAdapterXtermJS(element, bus)
     term["setOption"]("logLevel", "off");
     term.write("This is the serial console. Whatever you type or paste here will be sent to COM1");
 
-    term["onData"](function(data) {
+    const on_data_disposable = term["onData"](function(data) {
         for(let i = 0; i < data.length; i++)
         {
             bus.send("serial0-input", data.charCodeAt(i));
         }
     });
 
-    bus.register("serial0-output-char", function(chr)
+    bus.register("serial0-output-char", serial_char_handler, this);
+
+    this.destroy = function()
+    {
+        bus.unregister("serial0-output-char", serial_char_handler);
```

Comment:
This shouldn't be necessary. I think it should be removed, unless there is another reason to keep it.

Reply:
I think you're right. I added this while attempting to fix a memory leak in my project. My thought was that there would be a useless mutual reference between xterm.js and v86, so keeping one around could prevent the other from being garbage collected.

In hindsight, it probably would not make sense to keep _either_ of those instances around if they've both been "destroyed"/"disposed". I'll try removing it to see if my memory leak returns.

Reply:
I have confirmed that this line is not necessary and removed it.